src/main/java/pl/betoncraft/betonquest/conditions/DateCondition.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * BetonQuest - advanced quests for Bukkit
+ * Copyright (C) 2016  Jakub "Co0sh" Sapalski
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package pl.betoncraft.betonquest.conditions;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+import pl.betoncraft.betonquest.Instruction;
+import pl.betoncraft.betonquest.InstructionParseException;
+import pl.betoncraft.betonquest.QuestRuntimeException;
+import pl.betoncraft.betonquest.api.Condition;
+
+/**
+ * Returns true if given date is matched
+ *
+ * @author Archerymaister
+ */
+public class DateCondition extends Condition {
+
+	private Date startDate;
+	private Date endDate;
+	private SimpleDateFormat dateFormat;
+
+	public DateCondition(Instruction instruction) throws InstructionParseException {
+		super(instruction);
+		super.staticness = true;
+		super.persistent = true;
+
+		if(instruction.size() > 3){
+			throw new InstructionParseException("Too many arguments!");
+		}
+		if(instruction.size() < 2){
+			throw new InstructionParseException("not enough arguments!");
+		}
+		if (instruction.size() == 2) {
+			dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+			try {
+				startDate = cleanDate(dateFormat.parse(instruction.getPart(1)));
+			} catch (ParseException e) {
+				throw new InstructionParseException("Invalid date format!");
+			}
+		}else{
+			dateFormat = new SimpleDateFormat("dd.MM.yyyy-hh:mm:ss");
+			try {
+				startDate = TimeDate(dateFormat.parse(instruction.getPart(1)));
```

Comment:
So if you implement my suggestions above and below, this whole section of code will look like this:

```
try {
	dateFormat = new SimpleDateFormat("dd.MM.yyyy-hh:mm:ss");
	startDate = dateFormat.parse(instruction.next));
	endDate = dateFormat.parse(instruction.next));
} catch (ParseException e) {
	throw new InstructionParseException("Invalid date format!");
}
```

By the way, `dateFormat` variable could be a field now. Just a _final_ field, so it's easier to change it later without looking through the code.